wcfsetup/install/files/lib/system/background/job/EmailDeliveryBackgroundJob.class.php

```diff
@@ -59,6 +74,54 @@ public function __construct(Email $email, Mailbox $envelopeFrom, Mailbox $envelo
         $this->email = $email;
         $this->envelopeFrom = $envelopeFrom;
         $this->envelopeTo = $envelopeTo;
+
+        $this->emailLogEntryId = $this->createLog()->entryID;
+    }
+
+    /**
+     * Creates the email log entry.
+     */
+    private function createLog(): EmailLogEntry
+    {
+        return (new EmailLogEntryAction([], 'create', [
+            'data' => [
+                'time' => \TIME_NOW,
+                'messageID' => $this->email->getMessageID(),
+                'subject' => StringUtil::truncate($this->email->getSubject(), 30),
```

Comment:
```suggestion
                'subject' => StringUtil::truncate($this->email->getSubject(), 20),
```
Rationale: Trim to 20 characters to reduce the likelihood of exposing sensitive or confidential data in the mail subject.